scripts/release.ts

```diff
@@ -0,0 +1,192 @@
+  spawnSync(command, args, { stdio: 'inherit' });
+}
+
+function getCurrentCommit() {
+  return sh('git rev-parse HEAD');
+}
+
+function getCurrentBranch() {
+  return sh('git rev-parse --abbrev-ref HEAD');
+}
+
+function readLastPublishCommit() {
+  const commitHash = getCurrentCommit();
+  const commitMessage = sh('git log -1 --pretty=%B');
+  const tags = commitMessage.match(/@.*$/gm);
+
+  if (!tags) {
+    throw Error('The previous commit was not a publish commit. Cannot read tags!');
+  }
+
+  return { commitHash, commitMessage, tags };
+}
+
+async function undoLastCommit() {
+  const { commitHash, tags } = readLastPublishCommit();
+
+  console.log(c.red('The last publish commit will be deleted locally and on origin.'));
+  console.log(c.red('The following tags will also be deleted locally and on origin:\n'));
+  console.log(tags.join('\n'));
+  console.log('\n');
+
+  const yesContinue = await confirm({ message: 'Are you sure you want to continue?' });
+  if (!yesContinue) {
+    console.log(c.yellow('No action performed. Exiting...'));
+    process.exit(1);
+  }
+
+  console.log(c.green('Deleting tags and reverting commit...'));
+
+  const tagsOneLine = tags.join(' ');
+  sh(`git tag -d ${tagsOneLine}`);
+  sh(`git push origin --delete ${tagsOneLine}`);
+  console.log(c.green('Tags deleted.'));
+
+  sh(`git reset --hard ${commitHash}~1`);
+  sh(`git push --force origin ${getCurrentBranch()}`);
+  console.log(c.green('Publish commit deleted.'));
+}
+
+async function bumpVersions() {
+  console.log(c.green('Bumping package versions for release...'));
+  const preBumpHash = getCurrentCommit();
+  shI('./node_modules/.bin/lerna', ['version', '--no-push', '--exact']);
+  const postBumpHash = getCurrentCommit();
+
+  if (preBumpHash === postBumpHash) {
+    console.log(c.yellow('No version bump occurred. Exiting...'));
+    process.exit(1);
+  }
+
+  console.log(c.green('Package versions bumped successfully.'));
+  console.log(c.green('Updating versions.json for reference in docs...'));
+  appendVersions();
+  sh('git add -u');
+  sh('git commit --amend --no-edit');
+  console.log(c.green('Updated versions.json.'));
+
+  console.log(c.green('Pushing to origin...'));
+  sh(`git push --set-upstream origin ${getCurrentBranch()}`);
+  console.log(c.green('Pushed bump commit to origin.'));
+  sh(`git push origin ${readLastPublishCommit().tags.join(' ')}`);
+  console.log(c.green('Pushed tags to origin.'));
+}
+
+/**
+ * Assumes that we're on the publish commit when this function is called.
+ */
+async function bumpMain() {
+  const yes = await confirm({
+    message: `Would you like to create a pull request to bump versions on ${c.cyan('main')}?`,
+  });
+  if (!yes) {
+    console.log(c.green('Skipping version-bump pull request.'));
+    return;
+  }
+  console.log(c.green(`Creating a version-bump branch to merge into ${c.cyan('main')}...`));
+
+  const d = new Date().getDate();
+  const m = new Date().getMonth();
```

Comment:
do months start at 0? my PR says it was created last month

Reply:
[yup!](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getMonth)

> zero-based value (where zero indicates the first month of the year)

Reply:
doh! I also don't like how verbose it is. The original script used a shell command to format the date, and I'm tempted to do that here since it's one line and doesn't require a dependency like `moment`. Why is date formatting hard in JavaScript?

Reply:
Good catch